torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -1745,3 +1752,19 @@ def rref_tensor_is_owner(rref_var):
 
         res = rref_tensor_is_owner(rref_var)
         self.assertEqual(res, False)
+
+    @dist_init
+    def test_local_rref_with_script_module(self):
+        # create a local RRef that hold a ScriptModule
+        script_mod = MyScriptModule()
+        rref_module = rpc.RRef(script_mod)
+
+        @torch.jit.script
+        def rref_script_module(rref_module):
+            # type: (RRef[MyModuleInterface]) -> Tensor
+            return rref_module.to_here().forward()
+
+        # pass the local created RRef module to jit
+        res = rref_script_module(rref_module)
```

Comment:
this is pure local test? can we pass this RRef to remote workers?

Reply:
This right now could not be pass RRef to remote workers, it needs to work upon the jit pickler https://github.com/pytorch/pytorch/pull/32959, I will first test this locally as it is enough to distinguish from local test that we are holding a `ScriptModule` type in the RRef instead of a `PyObjectType`.

Reply:
hmm seems like even a local test with a remote rref need jit pickler/unpickler change. I will wait for that PR to land first and then land this.